src/sage/schemes/weighted_projective/weighted_projective_space.py

```diff
@@ -0,0 +1,418 @@
+        return SchemeHomset_points_weighted_projective_ring(*args, **kwds)
+
+    def point(self, v, check=True):
+        """
+        Create a point on this weighted projective space.
+
+        INPUT:
+
+        INPUT:
+
+        - ``v`` -- anything that defines a point
+
+        - ``check`` -- boolean (default: ``True``); whether
+          to check the defining data for consistency
+
+        OUTPUT: A point of this weighted projective space.
+
+        EXAMPLES::
+
+            sage: WP = WeightedProjectiveSpace(QQ, [1, 3, 1])
+            sage: WP.point([2, 3, 1])
+            (2 : 3 : 1)
+        """
+        from sage.rings.infinity import infinity
+
+        if v is infinity or (
+            isinstance(v, (list, tuple)) and len(v) == 1 and v[0] is infinity
+        ):
+            if self.dimension_relative() > 1:
+                raise ValueError("%s not well defined in dimension > 1" % v)
+            v = [1, 0]
+
+        return self.point_homset()(v, check=check)
+
+    def _point(self, *args, **kwds):
+        """
+        Construct a point.
+
+        For internal use only. See :mod:`morphism` for details.
+        """
+        from sage.schemes.weighted_projective.weighted_projective_point import SchemeMorphism_point_weighted_projective_ring
+        return SchemeMorphism_point_weighted_projective_ring(*args, **kwds)
+
+    def _repr_(self) -> str:
+        """
+        Return a string representation of this weighted projective space.
+
+        EXAMPLES::
+
+            sage: WeightedProjectiveSpace(Qp(5), [1, 3, 1], 'x')                        # needs sage.rings.padics
+            Weighted Projective Space of dimension 2 with weights (1, 3, 1)
+            over 5-adic Field with capped relative precision 20
+        """
+        return (
+            f"Weighted Projective Space of dimension {self.dimension_relative()} with weights"
+            f" {self.weights()} over {self.base_ring()}"
+        )
+
+    def change_ring(self, R):
+        r"""
+        Return a weighted projective space over ring ``R``.
+
+        INPUT:
+
+        - ``R`` -- commutative ring or morphism
+
+        OUTPUT: weighted projective space over ``R``
+
+        .. NOTE::
+
+            There is no need to have any relation between ``R`` and the base ring
+            of this space, if you want to have such a relation, use
+            ``self.base_extend(R)`` instead.
+
+        EXAMPLES::
+
+            sage: WP = WeightedProjectiveSpace([1, 3, 1], ZZ); WP
+            Weighted Projective Space of dimension 2 with weights (1, 3, 1) over Integer Ring
+            sage: WP.change_ring(QQ)
+            Weighted Projective Space of dimension 2 with weights (1, 3, 1) over Rational Field
+            sage: WP.change_ring(GF(5))
+            Weighted Projective Space of dimension 2 with weights (1, 3, 1) over Finite Field of size 5
+        """
+        if isinstance(R, Map):
```

Comment:
It would be nice to include an example of this in the docstring.

Reply:
actually I find this mathematically questionable. Map of rings induce map of their scheme in opposite direction, right.